tsdb/engine/tsm1/file_store.go

```diff
@@ -267,50 +267,21 @@ func (f *FileStore) Read(key string, t time.Time) ([]Value, error) {
 	return nil, nil
 }
 
-func (f *FileStore) Scan(key string, t time.Time, ascending bool) ([]Value, error) {
-	if ascending {
-		return f.next(key, t)
-	} else {
-		return f.prev(key, t)
-	}
-}
-
-func (f *FileStore) next(key string, t time.Time) ([]Value, error) {
+func (f *FileStore) KeyCursor(key string) *KeyCursor {
 	f.mu.RLock()
 	defer f.mu.RUnlock()
 
+	var locations []*location
 	for _, fd := range f.files {
-		// May have the key and time we are looking for so try to find
-		v, err := fd.Next(key, t)
-		if err != nil {
-			return nil, err
-		}
-
-		if len(v) > 0 {
-			return v, nil
+		for _, ie := range fd.Entries(key) {
+			locations = append(locations, &location{
+				r:     fd,
+				entry: ie,
+			})
 		}
 	}
-	return nil, nil
-}
-
-func (f *FileStore) prev(key string, t time.Time) ([]Value, error) {
-	f.mu.RLock()
-	defer f.mu.RUnlock()
-
-	for i := len(f.files) - 1; i >= 0; i-- {
-		fd := f.files[i]
-
-		// May have the key and time we are looking for so try to find
-		v, err := fd.Prev(key, t)
-		if err != nil {
-			return nil, err
-		}
 
-		if len(v) > 0 {
-			return v, nil
-		}
-	}
-	return nil, nil
+	return &KeyCursor{seeks: locations, buf: make([]Value, 1000)}
```

Comment:
OK, that would make sense. I just wanted to check it wasn't something like the decoding side of things *expected* up to 1000 keys, and would consider it an error otherwise.